MewsSystems/mews-flutter
# feat(optimus)!: [RND-103047] Add Notification showing and stacking [ignore-size]

#### Summary

RND-103047 Added showing and stacking notifications. 

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/9210422/178459960-c9e99b68-c268-4f34-94e9-c7419631beac.mp4


</details>

Breaking changes:

- `OptimusNotification` now requires `OptimusNotificationLink`, to assure both **link text** and **link callback** are provided. Or neither of them.

#### Testing steps

1. Open Storybook / Test in your project (wrap screen with OptimusNotificationsOverlay)
2. Try to show notifications inside this screen. 
3. Notifications should look and behave according to the Optimus Design System guidelines. 

#### Follow-up issues

- ~~RND-108205~~ fixed here.
- RND-114092 - fix code after a breaking change. 

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).
